pkg/util/serviceaccount.go

```diff
@@ -83,14 +83,34 @@ func WaitForServiceAccountSecretCreation(client kubeclient.Interface, asObj *cor
 			}
 			return false, fmt.Errorf("failed to retrieve service account(%s/%s) from cluster, err: %v", asObj.Namespace, asObj.Name, err)
 		}
-		clusterSecret, err = GetTargetSecret(client, serviceAccount.Secrets, corev1.SecretTypeServiceAccountToken, asObj.Namespace)
+		if len(serviceAccount.Secrets) == 0 {
+			clusterSecret, err = GetSecret(client, serviceAccount.Namespace, serviceAccount.Name)
+			if apierrors.IsNotFound(err) {
+				secret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: serviceAccount.Namespace,
+						Name:      serviceAccount.Name,
+						Annotations: map[string]string{
+							"kubernetes.io/service-account.name": serviceAccount.Name,
+						},
+					},
+					Type: corev1.SecretTypeServiceAccountToken,
+				}
+				clusterSecret, err = CreateSecret(client, secret)
+			}
+		} else {
+			clusterSecret, err = GetTargetSecret(client, serviceAccount.Secrets, corev1.SecretTypeServiceAccountToken, asObj.Namespace)
+		}
+
 		if err != nil {
 			if apierrors.IsNotFound(err) {
 				return false, nil
 			}
 			return false, err
 		}
-
+		if _, ok := clusterSecret.Data["token"]; !ok {
```

Comment:
after creating `clusterSecret`, you need to wait for the token controller to populate the Data["token"] field

Reply:
+1
Yes, it is the `token` we should waiting for, not the `secret`.

Reply:
I'm not sure if I understand correctly
wait already existed:
https://github.com/karmada-io/karmada/pull/1972/files#diff-b8f6d51c2457af24cffeb6f3b26e7b35f3df727ce4a851e082bfae84742f124aL78

so ,there just return false ,it will check again